pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -851,20 +885,40 @@ cdef class PyObjectHashTable(HashTable):
             val = values[i]
             hash(val)
 
-            if ((val != val or val is None) or
-                    (use_na_value and val == na_value)):
+            if ignore_na and ((val != val or val is None)
```

Comment:
this keyword is completely untested

Reply:
I would argue that it is (at least for `ignore_na=True), because all the tests for factorize need to run through this.

Do we explicitly test cython code? I can if necessary. This PR does the required preparations on cython level to continue with `return_inverse` later on. It will be excessively tested there.

Reply:
yes of course, you added a keyword.

Reply:
I am talking about the ignore_na keyword

Reply:
I know that I added this keyword to switch between `.unique`/`.factorize` behaviour.

Can you point me to where the cython hashtable code is tested? I could add some tests there. That's why I asked "Do we explicitly test cython code?", because I haven't come across it yet.

Otherwise, I would add copious tests for the behaviour as soon as the keyword makes its way into the pandas API, but this is just the preparation for that.

Reply:
@h-vetinari : IMO `tests/generic/test_generic.py` or `tests/generic/test_frame.py` could be good places to put any kinds of tests for this refactoring for now.

If it is possible to hit those keyword arguments from Python land, then do add some tests.

Reply:
let's drop the ignore_na for now. and add it when I see that you actually need it.